packages/library-legacy/src/programs/vote.ts

```diff
@@ -238,6 +250,41 @@ export class VoteInstruction {
     };
   }
 
+  /**
+   * Decode a compact update vote state instruction and retrieve the instruction params.
+   */
+  static decodeCompactUpdateVoteState(
+      instruction: TransactionInstruction,
+  ): CompactUpdateVoteStateParams {
+    this.checkProgramId(instruction.programId);
+    this.checkKeyLength(instruction.keys, 2);
+
+    const { voteStateUpdate: { lockoutOffsets, root, hash, timestampOption, timestamp }  } = decodeData(
+      VOTE_INSTRUCTION_LAYOUTS.CompactUpdateVoteState,
+      instruction.data,
+    );
+    let lockoutSlot = root;
+
+    return {
+      voteAccount: instruction.keys[0].pubkey,
+      voteAuthority: instruction.keys[1].pubkey,
+      voteStateUpdate: {
+        lockouts: lockoutOffsets.map(
```

Comment:
this transformation is made to be compliant with the output provided by `api.mainnet-beta.solana.com`, since node returns lockouts, while instruction for "compact" contains offset values.

Reply:
I see what you're going for here. For anyone else reading, this is required due to the custom Serde serialization method used for `CompactUpdateVoteState`, ie. [`serde_compact_vote_state_update`](https://github.com/solana-labs/solana/blob/27eff8408b7223bb3c4ab70523f8a8dca3ca6645/sdk/program/src/vote/state/mod.rs#L811).

Since this is required at a serialization level, and you'll use this twice (here and in the transaction function), why not roll this into a custom serialization layout?

You can define a class `MyLayout extends Layout<T>` and implement the required methods that can transform from the instruction input type to the proper byte layout, and back.

We did this in SPL Token JS to mimic Rust's `COption::<Pubkey>` and you can use that as an example.
https://github.com/solana-labs/solana-program-library/blob/d1c03cc6164ebdeebef19190de28b7a71f4e4cb6/token/js/src/serialization.ts#L5-L39